src/SprykerMiddleware/Zed/Process/Business/ProcessBusinessFactory.php

```diff
@@ -2,50 +2,212 @@
+    {
+        $postProcessHooks = $this->getProvidedDependency(ProcessDependencyProvider::MIDDLEWARE_POST_PROCESSOR_HOOKS_STACK);
+        return $postProcessHooks[$processName]?:[];
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ProcessSettingsTransfer $processSettingsTransfer
+     *
+     * @return \Iterator
+     */
+    protected function getProcessIterator(ProcessSettingsTransfer $processSettingsTransfer): Iterator
+    {
+        $iterators = $this->getProcessIteratorsList();
+        return $iterators[$processSettingsTransfer->getName()]($processSettingsTransfer->getIteratorSettings());
+    }
+
+    /**
+     * @return array
+     */
+    protected function getProcessIteratorsList(): array
+    {
+        return [];
+    }
+
+    /**
+     * @param \SprykerMiddleware\Zed\Process\Communication\Plugin\StagePluginInterface[] $stagePlugins
+     * @param \Psr\Log\LoggerInterface $logger
+     *
+     * @return \SprykerMiddleware\Zed\Process\Business\Pipeline\PipelineInterface
+     */
+    protected function createPipeline(array $stagePlugins, LoggerInterface $logger): PipelineInterface
+    {
+        return new Pipeline(
+            $this->createPipelineProcessor(),
+            $this->getStages($stagePlugins, $logger)
+        );
+    }
+
+    /**
+     * @param \SprykerMiddleware\Zed\Process\Communication\Plugin\StagePluginInterface[] $stagePlugins
+     * @param \Psr\Log\LoggerInterface $logger
+     *
+     * @return \SprykerMiddleware\Zed\Process\Business\Pipeline\Stage\StageInterface[]
+     */
+    protected function getStages(array $stagePlugins, LoggerInterface $logger): array
+    {
+        return $this->createStageListBuilder()
+            ->buildStageList($stagePlugins, $logger);
+    }
+
+    /**
+     * @return \SprykerMiddleware\Zed\Process\Business\Pipeline\Stage\StageListBuilderInterface
+     */
+    protected function createStageListBuilder(): StageListBuilderInterface
+    {
+        return new StageListBuilder();
+    }
+
+    /**
+     * @return \League\Pipeline\FingersCrossedProcessor
+     */
+    protected function createPipelineProcessor(): FingersCrossedProcessor
+    {
+        return new FingersCrossedProcessor();
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ProcessSettingsTransfer $processSettingsTransfer
+     *
+     * @return \Spryker\Shared\Log\Config\LoggerConfigInterface
+     */
+    protected function getProcessLoggerConfig(ProcessSettingsTransfer $processSettingsTransfer): LoggerConfigInterface
+    {
+        $configuration = $this->getProcessLoggerConfigList();
+        if (isset($configuration[$processSettingsTransfer->getName()])) {
+            return $configuration[$processSettingsTransfer->getName()]($processSettingsTransfer->getLoggerSettings());
+        }
+        return $this->createMiddlewareLogConfig($processSettingsTransfer->getLoggerSettings());
+    }
+
+    /**
+     * @return array
+     */
+    protected function getProcessLoggerConfigList(): array
```

Comment:
This should be configured in the `DependencyProvider`!

Reply:
Should be changed in future